include/seqan3/alphabet/nucleotide/rna4.hpp

```diff
@@ -56,113 +53,76 @@ namespace seqan3
-     */
-    //!\{
-    static const rna4 A;
-    static const rna4 C;
-    static const rna4 G;
-    static const rna4 U;
-    static const rna4 T;
-    static const rna4 UNKNOWN;
-    //!\}
-
-    /*!\name Read functions
-     * \{
-     */
-    //!\copydoc seqan3::dna4::to_char
-    constexpr char_type to_char() const noexcept
-    {
-        return value_to_char[static_cast<rank_type>(_value)];
-    }
+private:
+    //!\brief The base class.
+    using base_t = nucleotide_base<rna4, 4>;
 
-    //!\copydoc seqan3::dna4::complement
-    constexpr rna4 complement() const noexcept
-    {
-        return rna4{dna4::complement()};
-    }
-    //!\}
+    //!\brief Befriend seqan3::nucleotide_base.
+    friend base_t;
+    //!\cond \brief Befriend seqan3::alphabet_base.
+    friend base_t::base_t;
+    //!\endcond
 
-    /*!\name Write functions
+public:
+    /*!\name Constructors, destructor and assignment
      * \{
      */
-    //!\copydoc seqan3::dna4::assign_char
-    constexpr rna4 & assign_char(char_type const c) noexcept
-    {
-        using index_t = std::make_unsigned_t<char_type>;
-        _value = char_to_value[static_cast<index_t>(c)];
-        return *this;
-    }
-
-    //!\copydoc seqan3::dna4::assign_rank
-    constexpr rna4 & assign_rank(rank_type const c)
+    constexpr rna4() : base_t{} {}
+    constexpr rna4(rna4 const &) = default;
+    constexpr rna4(rna4 &&) = default;
+    constexpr rna4 & operator=(rna4 const &) = default;
+    constexpr rna4 & operator=(rna4 &&) = default;
+
+    //!\brief Allow implicit construction from dna/rna of the same size.
+    constexpr rna4(dna4 const & r)
     {
-        assert(c < value_size);
-        _value = static_cast<internal_type>(c);
-        return *this;
+        assign_rank(r.to_rank());
     }
     //!\}
 
-    /*!\name Conversion operators
-     * \{
-     */
-    //!\brief Implicit conversion between dna* and rna* of the same size.
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept and have the same \link value_size \endlink.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
-    //!\endcond
-    constexpr operator other_nucl_type() const noexcept
-    {
-        return other_nucl_type{_value};
-    }
-
-    //!\brief Explicit conversion to any other nucleotide alphabet (via char representation).
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type>
-    //!\endcond
-    explicit constexpr operator other_nucl_type() const noexcept
-    {
-        return detail::convert_through_char_representation<other_nucl_type, std::decay_t<decltype(*this)>>[to_rank()];
-    }
-    //!\}
+protected:
+    //!\privatesection
```

Comment:
do you need this?

Reply:
Well I don't want things to appear in the user doc.